docs/concepts/internationalization.md

```diff
@@ -0,0 +1,69 @@
+Internationalization and localization
+=====================================
+
+Internationalization, i18n for short, is a way of designing code so that it requires no changes to be adapted for various languages. Localization is the process of adapting displayed text to the user's language.
+
+I18n is implemented using _translation keys_. A translation key is a string that identifies a piece of displayable text in no specific language. For example, `tile.dirt.name` is the language key referring to the name of the Dirt block. This way, displayable text may be referenced with no concern for a specific language. The code requires no changes to be adapted in a new language.
+
+Localization will happen in the game's locale. In a Minecraft client the locale is specified by the language settings. On a dedicated server, the only supported locale is en_US.
+
+Language files
+--------------
+
+Language files are located by `assets/domain/lang/locale.lang` (e.g. the US English translation for `examplemod` would be `assets/examplemod/lang/en_us.lang`). Resource pack format 3 requires the locale name to be lowercased. The file format is simply lines of key-value pairs. Lines starting with `#` are treated as comments.
+
+```properties
+# items
+item.examplemod.example_item.name=Example Item Name
+
+# blocks
+tile.examplemod.example_block.name=Example Block Name
+
+# commands
+commands.examplemod.examplecommand.usage=/example <value>
+```
+
+A language file may contain escape sequences if the file starts with the comment `#PARSE_ESCAPES`.
```

Comment:
The comment turns on full blown properties syntax, not just escape sequences. The MC lang format isn't a subset of properties syntax (namely that files containing `:` in keys have different semantics in properties format, as the colon is a separator there), so I think elaborating a bit more on what this comment entails would be great.